referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
```

Comment:
I think I can get multiple birds with one stone:

Supplying a specific template when generating, e.g. ```./generic/tools/generate.py --target=debug``` or ```./generic/tools/generate.py --target=release```,
would help me move these sanity checks outside of the test case (into a ```sanity-checks.js``` just for debugging) and reduce the overall resources for each test when in production. What do you think? :-)